### ISSUE TITLE
Table level settings not passed to ClickHouse

### ISSUE THREAD

**Issue by jesperbagge** [ClickHouse/dbt-clickhouse#91]
Hi! I'm stuck trying to define a table engine setting through the model config. According to the [README.md](https://github.com/ClickHouse/dbt-clickhouse#model-configuration) it should be possible by defining the settings property with a dictionary, but I'm struggling on this one.

My model config looks something like this:

`{{ config(order_by='(Geography_Code,)', materialization='table', engine="MergeTree()", settings='({"allow_nullable_key":1})') }}`

The source is a table in a MySQL database. But I'm just getting the error:

`DB::Exception: Sorting key contains nullable columns, but merge tree setting `allow_nullable_key` is disabled.`

What am I missing?

**Comment by genzgd** [comment]
I'll look into why this isn't working.  In the meantime you could set it for all merge tree table in your ClickHouse config.xml if that's an option:

```
<merge_tree>
  <allow_nullable_key>1</allow_nullable_key>
</merge_tree>
```

**Comment by jesperbagge** [comment]
Thanks! Just using MySQL to get some "real data" for our PoC. Yeah, I realize that production hardened models are better off by always explicitly declaring datatypes. My data governance self will love me for it.

I managed to circumvent my error by using `coalesce(field, value-if-null)` to make sure that NULL will never enter my keys - which is a lot smarter that allowing it in the first place.

But I'm still curious to hear if the settings implementation is correct and I'm just not declaring it right or if there is room for improvement.

Cheers!

**Comment by genzgd** [comment]
I suspect there's something broken with the settings and we'll keep this open until we understand what went wrong.

**Comment by jesperbagge** [comment]
ClickHouse version is 22.8.1.17
dbt-core is 1.1.0
dbt-clickhouse is 1.1.7

**Comment by genzgd** [comment]
This was indeed broken.  It should be fixed in dbt-clickhouse 1.1.8.

The syntax for settings looks like this, btw:

```
 {{ config(order_by='(some_date, id, name)', engine='MergeTree()', materialized='table',
                     settings={'allow_nullable_key': 1}) }}
```

What you have above had some extra quotes/symbols.

**Comment by arniwesth** [comment]
I suggest adding this (back?) to https://github.com/ClickHouse/dbt-clickhouse#model-configuration - I had to go here to figure out how to define the settings property.
